### ISSUE TITLE
Refactor httpClient as singleton

### ISSUE THREAD

**Issue by jhorn70** [intacct/intacct-sdk-net#208]
I have been trying to troubleshoot an issue with task cancellation and stumbled on the below.  

https://www.aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?redirectedfrom=MSDN&view=net-6.0#Remarks

Looks like the best practice is to re-use httpClient across multiple requests.

**Comment by davidathompson** [comment]
Yes, that was the first question I head when digging through the code. A new HttpClient with every request? Yikes.....this could destabilize an app with enough volume.

**Comment by sfwatanabe** [comment]
To add this issue - it would be nice to allow the option to pass our own HttpClient instance the Offline/OnlineClient constructors

**Comment by davidathompson** [comment]
To further qualify this, a new HttpClient is actually fine, and is typically lightweight. And the logging wireup is useful here. What is actually at issue is: 

```
HttpClientHandler httpClientHandler = new HttpClientHandler()
```

done with each request. Allowing the client to supply the inner HttpMessageHandler could give you the best of all the worlds, allowing the same logging wireup as you have now. Or allow either/or to be supplied by the client if they want to simply supply the HttpClient. 

We typically pull our HttpClinets / HttpMessageHandlers from:

https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

Maybe add HttpMessageHandler and HttpClient properties to ClientConfig? That way you can fill in the HttpMessageHandler if you know what you are doing there or simply HttpClient if you want to simplify (or leave null to use the current methodology) ?

**Comment by davidathompson** [comment]
Actually looks like there is a MockHandler property of ClientConfig which in theory you could inherit from and override the SendAsync to delegate back to your own instance of HttpClientHandler. 

```
public MockHandler MockHandler;
```
I'm sure not the intention of the designers, but could easily be repurposed to manage the life cycle of HttpClient :)

**Comment by davidathompson** [comment]
Create pull request to address:

https://github.com/intacct/intacct-sdk-net/pull/232
